test/parallel/test-assert.js

```diff
@@ -645,6 +645,24 @@ try {
                      'Message incorrectly marked as generated');
 }
 
+// verify custom errors
+try {
+  assert.strictEqual(1, 2, new RangeError('my range'));
+} catch (e) {
+  assert.strictEqual(e.message, 'my range');
```

Comment:
If you store the `RangeError` in a variable, you can just assert that `e` equals it.